### PR TITLE
Add mania hidden & FadeIn mod configuration

### LINKED ISSUE #6208: osu!Mania Adjustable HD
**Describe the new feature:** 
This new feature would be a separate HD mod. Something that you could activate in the settings of the game possibly as some people wouldn't want this to clutter up their mod screen. 

I enjoy playing mania with HD. I also enjoy playing mania with a folder taped to my screen to hide most of my gameplay. Some people do enjoy having less screen area as it makes playing the game easier. Some enjoy having less screen area as it makes the game harder/more challenging. I am proposing an adjustable HD mod. I would prefer to have it be ranked, but give no additional score like the other mania mods. Reason for the mod being ranked instead of unranked is because one can achieve the same effect, but with a $1 school folder, some scotch tape, and a bit of effort taking it off whenever they actually want to use their computer.

For the feature itself it's essentially the same effect as HD, but the height of the "darking out" can be adjusted with a slider that is located in the same area where the background dim slider is located. I feel this would be a nice quality-of-life addition and also make it so I do not have to re-tape a folder to my screen to play this way. I feel that a lot of other mania players would also like having this mod for either fun or practical use.

### PR DESCRIPTION
**Pull request by Mogiiii**
* Closes #6208

Adds slider to configure how much of the screen the notes should be hidden for. Also adds `TestSceneManiaModHidden` with test cases at min, max, and default values.

### CODE REVIEW

File: osu.Game.Rulesets.Mania/Mods/ManiaModPlayfieldCover.cs [hunk 1]
```diff
@@ -22,6 +22,8 @@ public abstract class ManiaModPlayfieldCover : ModHidden, IApplicableToDrawableR
         /// </summary>
         protected abstract CoverExpandDirection ExpandDirection { get; }
 
+        public virtual float Coverage { get => 0.5f; }
```
Comment (bdach) [inline osu.Game.Rulesets.Mania/Mods/ManiaModPlayfieldCover.cs:25]: should be `protected`

File: osu.Game.Rulesets.Mania/Mods/ManiaModHidden.cs [hunk 2]
```diff
@@ -13,6 +15,16 @@ public class ManiaModHidden : ManiaModPlayfieldCover
         public override LocalisableString Description => @"Keys fade out before you hit them!";
         public override double ScoreMultiplier => 1;
 
+        [SettingSource("Coverage", "How much of the playfield notes should be hidden for.")]
```
Comment (bdach) [inline osu.Game.Rulesets.Mania/Mods/ManiaModHidden.cs:18]: This description doesn't parse very well to me - maybe this is better?

```suggestion
        [SettingSource("Coverage", "The proportion of playfield height that notes will be hidden for.")]
```

Also would consider displaying as percentage.

**Review by bdach — COMMENTED** [review]
Code quality inspections need work.

File: osu.Game.Rulesets.Mania/Mods/ManiaModHidden.cs (hunk 2 above)
Comment (Mogiiii) [reply osu.Game.Rulesets.Mania/Mods/ManiaModHidden.cs:18]: Thank you, that does sound better. I initially thought to use percentage as well, however all other mods are displayed as decimal so I think it is worth using decimal for the consistency.

**Review by bdach — DISMISSED** [review]
Technically looks sound to me, but I'll defer to a second opinion with regard to whether we want this setting at all and whether it's named and described correctly if yes.

File: osu.Game.Rulesets.Mania/Mods/ManiaModHidden.cs [hunk 3]
```diff
@@ -13,6 +15,17 @@ public class ManiaModHidden : ManiaModPlayfieldCover
         public override LocalisableString Description => @"Keys fade out before you hit them!";
         public override double ScoreMultiplier => 1;
 
+        [SettingSource("Coverage", "The proportion of playfield height that notes will be hidden for.")]
+        public BindableNumber<float> CoverageAmount { get; } = new BindableFloat(0.5f)
+        {
+            Precision = 0.01f,
```
[the remark this answers is not in the document]
Comment (Mogiiii) [reply osu.Game.Rulesets.Mania/Mods/ManiaModHidden.cs]: Yes, that is reasonable. I have made this change.

**Comment by peppy** [comment]
I've updated this to avoid the need for an in-between property, but not sure. Should we just add this customisation to FadeIn as well while we're here? It is probably easier to apply to both at once (osu-web etc) if we think it's going to be asked for in the future.

**Comment by Mogiiii** [comment]
> I've updated this to avoid the need for an in-between property, but not sure. Should we just add this customisation to FadeIn as well while we're here? It is probably easier to apply to both at once (osu-web etc) if we think it's going to be asked for in the future.

Yes, that sounds reasonable. I will make this change when I have some time, hopefully before tomorrow.

**Comment by Mogiiii** [comment]
I've moved the setting definition to `ManiaModPlayFieldCover` since it is common, but the definition of `Coverage` is still up to the inheriting classes. What I found to be a reasonable max for hidden is unplayable on fadein, so I have defined the limits differently per mod. To have equal limits would likely require some adjusting to how fading works which is outside the scope of this.

**Comment by peppy** [comment]
I'm not sure about the limits here but let's go with it for now and wait for people to play and complain.

**Review by peppy — APPROVED** [review]

**Pull request merged**